main/datalog.go

```diff
@@ -216,16 +216,17 @@ func bulkInsert(tbl string, db *sql.DB) (res sql.Result, err error) {
 
 	batchVals := insertBatchIfs[tbl]
 	for len(batchVals) > 0 {
-		i := int(0) // Maximum of 25 rows per INSERT statement.
+		i := int(0)
 		stmt := ""
 		vals := make([]interface{}, 0)
-		querySize := uint64(0) // Size of the query in bytes.
-		for len(batchVals) > 0 && i < 1000 && querySize < 750000 {
+		querySize := uint64(0)                                   // Size of the query in bytes.
+		for len(batchVals) > 0 && i < 30 && querySize < 750000 { // Maximum of 1000? columns for INSERT on default sqlite compile. trafficData is limiting function with 31 parameters.
```

Comment:
Or, use the much more flexible B+Trees and do away with SQL altogether.

Skypuppy
systems analyst, programmer, database designer and implementer, schema 
designer and implementer, and all-around goofy guy since 1980.



On 04/03/2016 08:41 AM, AvSquirrel wrote:
>
> In main/datalog.go 
> <https://github.com/cyoung/stratux/pull/365#discussion_r58310461>:
>
> >  		stmt := ""
> >  		vals := make([]interface{}, 0)
> > -		querySize := uint64(0) // Size of the query in bytes.
> > -		for len(batchVals) > 0 && i < 1000 && querySize < 750000 {
> > +		querySize := uint64(0)                                   // Size of the query in bytes.
> > +		for len(batchVals) > 0 && i < 30 && querySize < 750000 { // Maximum of 1000? columns for INSERT on default sqlite compile. trafficData is limiting function with 31 parameters.
>
> @cyoung <https://github.com/cyoung> -- I'll say up front that this 
> isn't going to be a large enough query size for |dump1090_terminal| or 
> |traffic| if you're tracking a decent number of targets, even at 0.5 
> second updates... but we literally can not set this larger than 30 or 
> 31 with our current traffic struct due to the 1000 column query limit.
>
> What do you think about rolling the |traffic| updates back to once per 
> second, and checking the |tbl| type to increase the query length for 
> |messages| and |dump1090_terminal|?
>
> (Or do we fork go-sqlite3 and custom compile to remove the 1000 column 
> limit?)
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub 
> <https://github.com/cyoung/stratux/pull/365/files/a546f18750e1ebf0f5d5086889ab3513ed0deb3a#r58310461>
>